File: my_project_cpp/src/number_publisher.cpp
// Nó que fica publicando no contador em uma determinada frequencia

#include "ros/ros.h"
#include "std_msgs/Float64.h"
#include "std_srvs/Empty.h"

class NumberPublisher{                            

    public:
        NumberPublisher(ros::NodeHandle *nh){     
            //Inicializar Variaveis
            number = 1.2;
            pub_interval = 1;
            reset_interval = 7;

            num_pub = nh->advertise<std_msgs::Float64>("/number", 10);
            timer_pub = nh->createTimer(ros::Duration(pub_interval), &NumberPublisher::timerCallback, this);
            client_reset = nh->serviceClient<std_srvs::Empty>("reset_counter");
            timer_reset = nh->createTimer(ros::Duration(reset_interval), &NumberPublisher::resetCallback, this);
        }

        void timerCallback(const ros::TimerEvent &event){
            std_msgs::Float64 msg;
            msg.data = this->number;
            num_pub.publish(msg);
        }

        void resetCallback(const ros::TimerEvent &event){
            std_srvs::Empty srv;
            ROS_INFO("Reset counter solicited");
            client_reset.call(srv);
        }

    private:
        double number;
        double pub_interval;
        double reset_interval;
        ros::Publisher num_pub;
        ros::Timer timer_pub;
        ros::ServiceClient client_reset;
        ros::Timer timer_reset;

};


int main(int argc, char **argv)
{
    ros::init(argc, argv, "number_publisher");     
    ros::NodeHandle nh;
    NumberPublisher number_publisher = NumberPublisher(&nh);    
    
    ros::spin();
    return 0;
}

File: src/my_project_py/scripts/number_publisher.py
#!/usr/bin/env python3

import rospy
import random
from std_msgs.msg import Float64
from std_srvs.srv import Empty

class NumberPublisher:

    def __init__(self):
        self.number = 3.1
        self.publisher_interval = 1.5
        self.reset_interval = 7

        self.number_publisher = rospy.Publisher("/number", Float64, queue_size=10)
        
        self.timer_pub = rospy.Timer(rospy.Duration(self.publisher_interval), self.timer_callback)
        self.timer_reset = rospy.Timer(rospy.Duration(self.reset_interval), self.reset_callback)
        
        self.client_reset = rospy.ServiceProxy("reset_counter", Empty)

        rospy.loginfo("Inicialized")

    def timer_callback(self, event):
        msg = Float64()
        msg.data = self.number
        self.number_publisher.publish(msg)
    
    def reset_callback(self, event):
        rospy.loginfo("Solicitação de reset da contagem em " + str(self.reset_interval) + " segundos.")
        self.reset_interval = random.randint(1, 20)
        self.timer_reset.shutdown()
        self.timer_reset = rospy.Timer(rospy.Duration(self.reset_interval), self.reset_callback)
        self.client_reset()
        

if __name__ == '__main__':
    try:
        rospy.init_node("number_publisher")
        NumberPublisher()
        rospy.spin()
    except rospy.ROSInterruptException:
        pass